mlflow/server/js/src/experiment-tracking/components/CompareRunArtifactView.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from 'react';
+import ShowArtifactPage from './artifact-view-components/ShowArtifactPage';
+import { RunInfoEntity } from '../types';
+import { useRunsArtifacts } from './experiment-page/hooks/useRunsArtifacts';
+import { getCommonArtifacts } from './experiment-page/utils/getCommonArtifacts';
+import { CompareRunArtifactViewSidebar } from './CompareRunArtifactViewSidebar';
+import { useDesignSystemTheme } from '@databricks/design-system';
+import './CompareRunArtifactView.css';
+import { FormattedMessage } from 'react-intl';
+
+export const CompareRunArtifactView = ({
+  runUuids,
+  runInfos,
+  colWidth,
+}: {
+  runUuids: string[];
+  runInfos: RunInfoEntity[];
+  colWidth: number;
+}) => {
+  const { theme } = useDesignSystemTheme();
+  const [artifactPath, setArtifactPath] = useState<string>();
+
+  const { artifactsKeyedByRun } = useRunsArtifacts(runUuids);
+  const commonArtifacts = getCommonArtifacts(artifactsKeyedByRun);
+
+  if (commonArtifacts.length === 0) {
+    return (
+      <h2>
+        <FormattedMessage
+          defaultMessage="No common artifacts to display."
+          description="Text shown when there are no common artifacts between the runs"
+        />
+      </h2>
+    );
+  }
+  return (
+    <div className="artifact-container">
+      <CompareRunArtifactViewSidebar artifacts={commonArtifacts} onSelectArtifact={setArtifactPath} />
```

Comment:
i think we can reuse [ArtifactViewTree](https://github.com/mlflow/mlflow/blob/532381db1f357c18e2d616faf179cf6e35744e15/mlflow/server/js/src/experiment-tracking/components/ArtifactViewTree.tsx#L22) here, to reduce code duplication

all that's required should be to massage the data into the correct format, it appears only 3 keys are required (`id`, `active`, and `name`):

```suggestion
<ArtifactViewTree
  data={commonArtifacts.map((path: string) => ({
    id: path,
    active: artifactPath === path,
    name: getBasename(path),
  }))}
  onToggleTreebeard={({ id }) => setArtifactPath(id)}
/>
```

Reply:
Originally, I changed the styles in order to get the images to align with the selected run columns (so I had to remove the width properties and improve the flex). I also added the border because the `ShowArtifactPage` itself has a bottom border and it doesn't look too good when our container height is the entire screen.

As for why I set the container height to `100vh`, if we don't do that, then when the user clicks the container ends up resizing based on the height of the artifact (causing the page to scroll up and only show the image partially) which I want to avoid from a UX perspective. :)

### Without the border
![image](https://github.com/user-attachments/assets/83ae5ac0-ea66-4c31-8b6b-44e4fd3ab26c)
![image](https://github.com/user-attachments/assets/002ab6f3-148c-4bb7-b203-c51b58c1c23e)

### With the border
![image](https://github.com/user-attachments/assets/b4e78938-33ff-4f6a-a95f-9c5af135f7b7)

Reply:
I didn't have any issues refactoring to use the `ArtifactViewTree`, though!